File: seed.js
#!/usr/bin/env node
/* Seed SQLite từ dữ liệu Firebase export (User.json, Admin.json) */

const Database = require('better-sqlite3');
const path = require('path');
const fs = require('fs');

const db = new Database(path.join(__dirname, 'data.db'));
db.pragma('journal_mode=WAL');

// Tạo bảng nếu chưa có (giống server.js)
db.exec(`
  CREATE TABLE IF NOT EXISTS users (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    tentk TEXT UNIQUE NOT NULL,
    pass TEXT NOT NULL,
    sdt TEXT DEFAULT '',
    hoten TEXT DEFAULT '',
    permisson INTEGER DEFAULT 0
  );
  CREATE TABLE IF NOT EXISTS admins (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    username TEXT UNIQUE NOT NULL,
    matkhau TEXT NOT NULL
  );
  CREATE TABLE IF NOT EXISTS config (
    key TEXT PRIMARY KEY,
    value TEXT DEFAULT ''
  );
`);

// --- Seed users ---
const userPath = path.join(__dirname, 'User.json');
if (fs.existsSync(userPath)) {
  const users = JSON.parse(fs.readFileSync(userPath, 'utf-8'));
  const insert = db.prepare('INSERT OR IGNORE INTO users (tentk, pass, sdt, hoten, permisson) VALUES (?, ?, ?, ?, ?)');
  let count = 0;
  const tx = db.transaction(() => {
    for (const u of users) {
      const info = insert.run(u.tentk, u.pass, u.sdt || '', u.hoten || '', parseInt(u.permisson) || 0);
      if (info.changes > 0) count++;
    }
  });
  tx();
  console.log(`Users: ${count} imported (${users.length} total in file)`);
} else {
  console.log('User.json not found, skipping users');
}

// --- Seed admins ---
const adminPath = path.join(__dirname, 'Admin.json');
if (fs.existsSync(adminPath)) {
  const admins = JSON.parse(fs.readFileSync(adminPath, 'utf-8'));
  const insert = db.prepare('INSERT OR IGNORE INTO admins (username, matkhau) VALUES (?, ?)');
  let count = 0;
  const tx = db.transaction(() => {
    for (const a of admins) {
      const info = insert.run(a.username, a.matkhau);
      if (info.changes > 0) count++;
    }
  });
  tx();
  console.log(`Admins: ${count} imported (${admins.length} total in file)`);
} else {
  console.log('Admin.json not found, skipping admins');
}

// Seed config mặc định nếu chưa có
const cfgCount = db.prepare('SELECT COUNT(*) as c FROM config').get().c;
if (cfgCount === 0) {
  db.prepare('INSERT INTO config (key, value) VALUES (?, ?)').run('webviewUrl', '');
  db.prepare('INSERT INTO config (key, value) VALUES (?, ?)').run('linkContact', '');
  db.prepare('INSERT INTO config (key, value) VALUES (?, ?)').run('powerby', '');
  console.log('Config: defaults seeded');
}

db.close();
console.log('Done.');

File: seed.py
#!/usr/bin/env python3
"""Seed SQLite từ Firebase export JSON files."""
import json, os, sqlite3

DB_PATH = os.path.join(os.path.dirname(__file__), 'data.db')

conn = sqlite3.connect(DB_PATH)
conn.executescript('''
  PRAGMA journal_mode=WAL;
  CREATE TABLE IF NOT EXISTS users (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    tentk TEXT UNIQUE NOT NULL,
    pass TEXT NOT NULL,
    sdt TEXT DEFAULT '',
    hoten TEXT DEFAULT '',
    permisson INTEGER DEFAULT 0
  );
  CREATE TABLE IF NOT EXISTS admins (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    username TEXT UNIQUE NOT NULL,
    matkhau TEXT NOT NULL
  );
  CREATE TABLE IF NOT EXISTS config (
    key TEXT PRIMARY KEY,
    value TEXT DEFAULT ''
  );
''')

# Users
user_path = os.path.join(os.path.dirname(__file__), 'User.json')
if os.path.exists(user_path):
    with open(user_path) as f:
        users = json.load(f)
    count = 0
    for u in users:
        try:
            conn.execute(
                'INSERT OR IGNORE INTO users (tentk, pass, sdt, hoten, permisson) VALUES (?, ?, ?, ?, ?)',
                (u['tentk'], u['pass'], u.get('sdt', ''), u.get('hoten', ''), int(u.get('permisson', 0)))
            )
            if conn.total_changes > 0:
                count += 1
        except Exception as e:
            print(f"  Skip user {u.get('tentk')}: {e}")
    print(f"Users: {count} imported ({len(users)} in file)")
    conn.commit()

# Admins
admin_path = os.path.join(os.path.dirname(__file__), 'Admin.json')
if os.path.exists(admin_path):
    with open(admin_path) as f:
        admins = json.load(f)
    count = 0
    for a in admins:
        try:
            conn.execute('INSERT OR IGNORE INTO admins (username, matkhau) VALUES (?, ?)', (a['username'], a['matkhau']))
            if conn.total_changes > 0:
                count += 1
        except Exception as e:
            print(f"  Skip admin {a.get('username')}: {e}")
    print(f"Admins: {count} imported ({len(admins)} in file)")

# Config defaults
cfg_count = conn.execute('SELECT COUNT(*) FROM config').fetchone()[0]
if cfg_count == 0:
    for k in ('webviewUrl', 'linkContact', 'powerby'):
        conn.execute('INSERT INTO config (key, value) VALUES (?, ?)', (k, ''))
    print("Config: defaults seeded")

conn.commit()
conn.close()
print('Done.')
